core/src/main/java/ch/cyberduck/core/Local.java

```diff
@@ -359,7 +359,33 @@ public void rename(final Local renamed) throws AccessDeniedException {
     }
 
     public void copy(final Local copy) throws AccessDeniedException {
-        this.copy(copy, new CopyOptions());
+        this.channelCopy(copy, new CopyOptions());
+    }
+
+    public void channelCopy(final Local copy, final CopyOptions options) throws AccessDeniedException {
+        if(copy.equals(this)) {
+            log.warn(String.format("%s and %s are identical. Not copied.", this.getName(), copy.getName()));
+        }
+        else {
+            if(log.isDebugEnabled()) {
+                log.debug(String.format("Copy to %s with options %s", copy, options));
+            }
+            FileChannel c1 = null;
+            FileChannel c2 = null;
+            try {
+                c1 = this.getRawChannelInputStream();
+                c2 = copy.getRawChannelOutputStream(options.append);
+                c2.transferFrom(c1, c2.size(), c1.size());
+            }
+            catch(IOException e) {
+                throw new LocalAccessDeniedException(MessageFormat.format(
+                    LocaleFactory.localizedString("Cannot copy {0}", "Error"), this.getName()), e);
+            }
+            finally {
+                IOUtils.closeQuietly(c1);
+                IOUtils.closeQuietly(c2);
+            }
+        }
     }
 
     public void copy(final Local copy, final CopyOptions options) throws AccessDeniedException {
```

Comment:
I see no reason to keep this old implementation but would just replace it with the new implementation using the `FileChannel`.

Reply:
fixed